external/umdutils/verifier.js

```diff
@@ -0,0 +1,488 @@
+ *   } else if (typeof exports !== 'undefined') {
+ *     factory(exports, require('../shared/util.js'), require('./webgl.js'));
+ *   } else {
+ *     factory((root.pdfjsDisplayPatternHelper = {}), root.pdfjsSharedUtil,
+ *       root.pdfjsDisplayWebGL);
+ *   }
+ * }(this, function (exports, sharedUtil, displayWebGL) {
+ *
+ */
+
+var fs = require('fs');
+var path = require('path');
+
+/**
+ * Parses PDF.js UMD header.
+ * @param {string} filePath PDF.js JavaScript file path.
+ * @returns {{amdId: *, amdImports: Array, cjsRequires: Array, jsRootName: *,
+ *   jsImports: Array, imports: Array, importedNames: Array,
+ *   exportedNames: Array, body: *}}
+ */
+function parseUmd(filePath) {
+  var jscode = fs.readFileSync(filePath).toString();
+  // Extracts header and body.
+  var umdStart = '\\(function\\s\\(root,\\sfactory\\)\\s\\{';
+  var umdImports = '\\}\\(this,\\sfunction\\s\\(exports\\b';
+  var umdBody = '\\)\\s\\{';
+  var umdEnd = '\\}\\)\\);\\s*$';
+  var m, re;
+  m = new RegExp(umdStart + '([\\s\\S]*?)' + umdImports + '([\\s\\S]*?)' +
+    umdBody + '([\\s\\S]*?)' + umdEnd).exec(jscode);
+  if (!m) {
+    throw new Error('UMD was not found');
+  }
+  var header = m[1];
+  var imports = m[2].replace(/\s+/g, '').split(',');
+  imports.shift(); // avoiding only-export case
+  var body = m[3];
+
+  // Extracts AMD definitions.
+  var amdMatch = /\bdefine\('([^']*)',\s\[([^\]]*)\],\s+factory\);/.
+    exec(header);
+  if (!amdMatch) {
+    throw new Error('AMD call was not found');
+  }
+  var amdId = amdMatch[1];
+  var amdImports = amdMatch[2].replace(/[\s']+/g, '').split(',');
+  if (amdImports[0] !== 'exports') {
+    throw new Error('exports expected first at AMD call');
+  }
+  amdImports.shift();
+
+  // Extracts CommonJS definitions.
+  var cjsMatch = /\bfactory\(exports((?:,\s+require\([^\)]+\))*)\);/.
+    exec(header);
+  if (!cjsMatch) {
+    throw new Error('CommonJS call was not found');
+  }
+  var cjsRequires = cjsMatch[1].replace(/\s+/g, ' ').trim().
+    replace(/\s*require\('([^']*)'\)/g, '$1').split(',');
+  cjsRequires.shift();
+  var jsMatch = /\bfactory\(\(root\.(\S+)\s=\s\{\}\)((?:,\s+root\.\S+)*)\);/.
+    exec(header);
+  if (!jsMatch) {
+    throw new Error('Regular JS call was not found');
+  }
+
+  // Extracts global object properties definitions.
+  var jsRootName = jsMatch[1];
+  var jsImports = jsMatch[2].replace(/\s+/g, '').split(',');
+  jsImports.shift();
+
+  // Scans for imports usages in the body.
+  var importedNames = [];
+  if (imports.length > 0) {
+    re = new RegExp('\\b(' + imports.join('|') + ')\\.(\\w+)', 'g');
+    while ((m = re.exec(body))) {
+      importedNames.push(m[0]);
+    }
+  }
+  importedNames.sort();
+  for (var i = importedNames.length - 1; i > 0; i--) {
+    if (importedNames[i - 1] === importedNames[i]) {
+      importedNames.splice(i, 1);
+    }
+  }
+  // Scans for exports definitions in the body.
+  var exportedNames = [];
+  re = /\bexports.(\w+)\s*=\s/g;
```

Comment:
While testing the PR locally, I wanted to check that the validator failed if an `exports.<name>` wasn't present. In order to do that, I initially tested to just comment out a random `exports.<name>` line. To my surprise, that didn't fail, which lead me this line.

So my question is: Should this regexp be changed to also check that the line with `exports.<name>` isn't commented out, since that would cause a runtime failure that the validator currently doesn't catch?